### PR TITLE
Add ability to delete avatar or header picture via the API

### LINKED ISSUE #18921: Allow deleting your avatar and header via the /api endpoints
### Pitch

Either new endpoints, or modify `PATCH /api/v1/accounts/update_credentials` in the credentials_controller.rb to accept empty or special values for `avatar` and `header` parameters

### Motivation

Currently this appears to be done via `DELETE /settings/profile/pictures/avatar` or `DELETE /settings/profile/pictures/header` but it should also be accessible via the Mastodon API officially and not just internally.

### PR DESCRIPTION
**Pull request by danielmbrasil**
This PR adds a new endpoint to the API which allows deleting the avatar or header picture of an account's profile.

**New endpoint:** `DELETE /api/v1/profile/:image`. Where `image` can be either `avatar` or `header`.
**Scope:** `write:accounts`.

If accepted, should close  #18921.

EDIT:

Documentation PR: mastodon/documentation#1238.

### CODE REVIEW

File: app/controllers/api/v1/accounts/pictures_controller.rb [hunk 1]
```diff
@@ -0,0 +1,26 @@
+# frozen_string_literal: true
+
+class Api::V1::Accounts::PicturesController < Api::BaseController
+  before_action -> { doorkeeper_authorize! :write, :'write:accounts' }
+  before_action :require_user!
+  before_action :set_picture
+
+  def destroy
+    raise(Mastodon::InvalidParameterError, 'Picture must be either "avatar" or "header"') unless valid_picture?
```
Comment (danielmbrasil) [inline app/controllers/api/v1/accounts/pictures_controller.rb:9]: Not sure if it's a good way to handle this.

**Comment by renchap** [comment]
This will need to be added to the documentation I assume?

**Comment by andypiper** [comment]
Good point @renchap - as a side point, we should add a note to CONTRIBUTING.md that API additions are expected to have a documentation PR to accompanying them.

**Comment by danielmbrasil** [comment]
> This will need to be added to the documentation I assume?

Working on it 👍🏻 

> we should add a note to CONTRIBUTING.md that API additions are expected to have a documentation PR to accompany them

Good point, I didn't know about it.

**Comment by danielmbrasil** [comment]
>This will need to be added to the documentation I assume?
> > Working on it 👍🏻

Done.

**Comment by danielmbrasil** [comment]
Any updates on this? Anything to be added in order to get this reviewed?

File: app/controllers/api/v1/accounts/pictures_controller.rb [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+class Api::V1::Accounts::PicturesController < Api::BaseController
+  before_action -> { doorkeeper_authorize! :write, :'write:accounts' }
+  before_action :require_user!
+  before_action :set_picture
+  before_action :validate_picture_param
+
+  def destroy
+    @account = current_account
+    UpdateAccountService.new.call(@account, { @picture => nil, "#{@picture}_remote_url" => '' }, raise_error: true)
```
Comment (ClearlyClaire) [inline app/controllers/api/v1/accounts/pictures_controller.rb:11]: The `#{@picture}_remote_url` bit should not be relevant for local users, which the endpoint is for. I'd rather remove it.

```suggestion
    UpdateAccountService.new.call(@account, { @picture => nil }, raise_error: true)
```

**Review by ClearlyClaire — CHANGES_REQUESTED** [review]
Thanks for your contribution! Apart from my inline comment, the code itself looks good to me!

I'm less sure about the API endpoint names, but I don't really have a better suggestion, so I won't block on it, but if anyone has an opinion on it that would be helpful!

**Comment by trwnh** [comment]
Mastodon API compatible implementations often extend off of `PATCH /api/v1/accounts/update_credentials`, which accepts explicit null values for `avatar` and `header`. This is probably not the best way to do it, but it is the most similar to current practice. If coming up with a new path, I'd propose `/api/v1/profile` as the stub for it. This would result in the following potential methods:

- `DELETE /api/v1/profile/avatar` (and the corresponding `PUT /api/v1/profile/avatar`)
- `DELETE /api/v1/profile/header` (and the corresponding `PUT /api/v1/profile/header`)

**Comment by ClearlyClaire** [comment]
I think those are good suggestions. @danielmbrasil can you use those endpoint names? (not necessarily supporting `PUT` for now)

**Review by ClearlyClaire — APPROVED** [review]

**Pull request merged**